pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -144,9 +200,24 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			logger.Info("Inspecting Deploymen object...")
 
 			for i, c := range deploymentObj.Spec.Template.Spec.Containers {
-				logger.Info("Adding EnvFrom to container")
-				deploymentObj.Spec.Template.Spec.Containers[i].EnvFrom = r.appendEnvFrom(
-					c.EnvFrom, instance.GetName())
+				if len(retriever.Data) > 0 {
+					logger.Info("Adding EnvFrom to container")
+					deploymentObj.Spec.Template.Spec.Containers[i].EnvFrom = r.appendEnvFrom(
+						c.EnvFrom, instance.GetName())
+				}
+				if len(retriever.VolumeKeys) > 0 {
+					logger.Info("Adding VolumeMounts to container")
+					mountPath := "/var/data"
+					if instance.Spec.MountPathPrefix != "" {
+						mountPath = instance.Spec.MountPathPrefix
+					}
+					deploymentObj.Spec.Template.Spec.Containers[i].VolumeMounts = r.appendVolumeMounts(
+						c.VolumeMounts, instance.GetName(), mountPath)
+					logger.Info("Adding Volumes to pod")
+					deploymentObj.Spec.Template.Spec.Volumes = r.appendVolumes(
+						deploymentObj.Spec.Template.Spec.Volumes, retriever.Data, retriever.VolumeKeys, instance.GetName(), instance.GetName()+"-volumemount")
```

Comment:
Isn't the name of the created intermediate secret `instance.GetName()` instead of `instance.GetName()+"-volumemount"` ?

Reply:
Done. 32517eaefaab24f44ce41fc73bbe85604aac6faa